src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,112 @@
+/* eslint-disable default-case */
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { Input } from '../../types/Input';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (
+    newMovie: Movie,
+    event: React.FormEvent,
+    clearFunction: CallableFunction,
+  ) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [imgUrlError, setImgUrlError] = useState(false);
```

Comment:
Rename all [boolean variables](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig) 
```suggestion
  const [isImgUrlError, setIsImgUrlError] = useState(false);
```